Service/XxllncToZGWZaakService.php

```diff
@@ -0,0 +1,378 @@
+    public function getZaakTypeByExtId(string $caseTypeId)
+    {
+        // Find existing zaaktype
+        $zaakTypeSync = $this->synchronizationService->findSyncBySource($this->xxllncAPI, $this->zaakTypeSchema, $caseTypeId);
+        if ($zaakTypeSync && $zaakTypeSync->getObject()) {
+            isset($this->io) && $this->io->info("Found a existing zaaktype with sourceId: $caseTypeId and gateway id: {$zaakTypeSync->getObject()->getId()->toString()}");
+
+            return $zaakTypeSync->getObject();
+        }
+
+        // Fetch and create new zaaktype
+        $zaakTypeObject = $this->xxllncToZGWZaakTypeService->getZaakType($caseTypeId);
+        if ($zaakTypeObject) {
+            return $zaakTypeObject;
+        }
+
+        isset($this->io) && $this->io->error("Could not find or create ZaakType for id: $caseTypeId");
+
+        return null;
+    } // end getZaakTypeByExtId
+
+    /**
+     * Makes sure this action has all the gateway objects it needs.
+     *
+     * @return bool false if some object couldn't be fetched
+     */
+    private function getRequiredGatewayObjects(): bool
+    {
+        // Get xxllnc source
+        if (!isset($this->xxllncAPI) && !$this->xxllncAPI = $this->sourceRepo->findOneBy(['location' => 'https://development.zaaksysteem.nl/api/v1'])) {
+            isset($this->io) && $this->io->error('Could not find Source: Xxllnc API');
+
+            return false;
+        }
+
+        // Get ZaakType schema
+        if (!isset($this->zaakTypeSchema) && !$this->zaakTypeSchema = $this->schemaRepo->findOneBy(['name' => 'ZaakType'])) {
+            isset($this->io) && $this->io->error('Could not find Schema: ZaakType');
+
+            return false;
+        }
+
+        // Get Zaak schema
+        if (!isset($this->zaakSchema) && !$this->zaakSchema = $this->schemaRepo->findOneBy(['name' => 'Zaak'])) {
+            isset($this->io) && $this->io->error('Could not find Schema: Zaak');
+
+            return false;
+        }
+
+        if (!isset($this->caseMapping) && !$this->caseMapping = $this->mappingRepo->findOneBy(['reference' => 'https://development.zaaksysteem.nl/api/v1/case'])) {
+            isset($this->io) && $this->io->error('No mapping found for https://development.zaaksysteem.nl/api/v1/case');
+
+            return null;
+        }
+
+        return true;
+    } // end getRequiredGatewayObjects
+
+    /**
+     * Sets default values.
+     *
+     * @param array $zaakArray
+     *
+     * @return array $zaakArray
+     */
+    private function setDefaultValues($zaakArray)
+    {
+        foreach ($this->skeletonIn as $key => $data) {
+            if (!isset($zaakArray[$key])) {
+                $zaakArray[$key] = $data;
+            }
+        }
+
+        return $zaakArray;
+    }// end setDefaultValues
+
+    /**
+     * Creates or updates a case to zaak.
+     *
+     * @param array $case Case from the Xxllnc API
+     *
+     * @return void|null
+     */
+    public function caseToZaak(array $case)
```

Comment:
This is a big function, could be smaller

Reply:
Add a todo for this

Reply:
![image](https://user-images.githubusercontent.com/57346398/224293428-df5d516b-2843-4c25-abc9-c062d88b2c07.png)
opgeschoond